### ISSUE TITLE
Sourcery does not provide code quality metrics for functions with assignment expressions

### ISSUE THREAD

**Issue by br3ndonland** [sourcery-ai/sourcery#130]
### Issue description or question

[Python 3.8 introduced assignment expressions](https://docs.python.org/3/whatsnew/3.8.html), which are created with the "walrus operator" (`:=`). Sourcery does not appear to provide code quality metrics for functions that contain assignment expressions.

To see this, we can adapt the simple example from the Python 3.8 release notes linked above.

```py
def limit_to_ten_no_assignment(a: list) -> None:
    """Sourcery provides code quality metrics for this function."""
    if len(a) > 10:
        print(f"List is too long ({len(a)} elements, expected <= 10)")


def limit_to_ten_with_assignment(a: list) -> None:
    """Sourcery does not provide code quality metrics for this function."""
    if (n := len(a)) > 10:
        print(f"List is too long ({n} elements, expected <= 10)")

```

Sourcery provides code quality metrics for the function without an assignment expression.

<img width="600" alt="sourcery-issue-walrus-01" src="https://user-images.githubusercontent.com/26674818/113498514-8977a380-94db-11eb-8bc0-5fa80da572e9.png">

Sourcery does not provide code quality metrics for the function with an assignment expression.

<img width="600" alt="sourcery-issue-walrus-02" src="https://user-images.githubusercontent.com/26674818/113498516-92687500-94db-11eb-96a9-5ca3a89c2b20.png">

It's worth considering an approach to assessing code quality when assignment expressions are used. The Python 3.8 release notes provide the following suggestion:

> Try to limit use of the walrus operator to clean cases that reduce complexity and improve readability.

How might assignment expressions impact working memory and other Sourcery metrics?

### Sourcery Version

0.8.8

### Code editor or IDE name and version

VSCode 1.55-insider (the same results have been observed consistently on all versions of Sourcery and VSCode I've used on macOS and Ubuntu)

### OS name and version

- OSX/macOS 11
- Ubuntu Linux 20.04 (LTS) x86_64

**Comment by Hellebore** [comment]
Thanks for raising! 

We have some other priorities for the next few weeks, but after that will definitely be looking to get this working.

**Comment by Hellebore** [comment]
Fixed in 0.9.5
